### PR TITLE
ADD deleteProfile

### PR DESCRIPTION
**Pull request by sarovin**
@n6g7 Can you help me with jest test?

### CODE REVIEW

File: src/auth.test.js [hunk 1]
```diff
@@ -30,7 +30,8 @@ describe('auth', () => {
       unlink: jest.fn(),
       updateEmail: jest.fn(),
       updatePassword: jest.fn(),
-      updateProfile: jest.fn()
+      updateProfile: jest.fn(),
+      deleteProfile: jest.fn()
```
Comment (n6g7) [inline src/auth.test.js]: I think your problem is here: [the method is called `delete` and not `deleteProfile`](https://firebase.google.com/docs/reference/js/firebase.User#delete). ;)

File: README.md [hunk 2]
```diff
@@ -73,7 +71,8 @@ Make sure your client provides a implementation of [`fetch`](https://developer.m
 - [`*auth.unlink(authProvider)`](https://redux-saga-firebase.js.org/reference/auth#unlink)
 - [`*auth.updateEmail(email)`](https://redux-saga-firebase.js.org/reference/auth#updateEmail)
 - [`*auth.updatePassword(password)`](https://redux-saga-firebase.js.org/reference/auth#updatePassword)
-- [`*auth.updateProfile(profile)`](https://redux-saga-firebase.js.org/reference/auth#updateProfile)
+- [`*auth.updateProfile(profile)`](https://redux-saga-firebase.js.org/reference/auth#deleteProfile)
```
Comment (n6g7) [inline README.md]: I don't think you meant to change this link. ;)

File: README.md [hunk 3]
```diff
@@ -27,23 +27,21 @@ import '@firebase/firestore'; // 👈 If you're using firestore
...
 ```js
 function* syncSaga() {
-  yield fork(
-    reduxSagaFirebase.database.sync,
-    'todos',
-    { successActionCreator: syncTodos }
-  );
+  yield fork(reduxSagaFirebase.database.sync, 'todos', {
+    successActionCreator: syncTodos,
+  });
```
Comment (n6g7) [inline README.md]: can you undo the changes above please?

File: README.md [hunk 4]
```diff
@@ -73,7 +71,8 @@ Make sure your client provides a implementation of [`fetch`](https://developer.m
 - [`*auth.unlink(authProvider)`](https://redux-saga-firebase.js.org/reference/auth#unlink)
 - [`*auth.updateEmail(email)`](https://redux-saga-firebase.js.org/reference/auth#updateEmail)
 - [`*auth.updatePassword(password)`](https://redux-saga-firebase.js.org/reference/auth#updatePassword)
-- [`*auth.updateProfile(profile)`](https://redux-saga-firebase.js.org/reference/auth#updateProfile)
+- [`*auth.updateProfile(profile)`](https://redux-saga-firebase.js.org/reference/auth#deleteProfile)
+- [`*auth.deleteProfile()`](https://redux-saga-firebase.js.org/reference/auth#deleteProfile)
```
Comment (n6g7) [inline README.md]: can you keep the methods ordered alphabetically please?

File: docs/reference/dev/auth.md [hunk 5]
```diff
@@ -502,4 +502,22 @@ methods:
...
+      function* deleteProfileSaga() {
+        try {
+          yield call(rsf.auth.deleteProfile);
+          yield put(deleteProfileSuccess());
+        }
+        catch(error) {
+          yield put(deleteProfileFailure(error));
+        }
+      }
+      ```
```
Comment (n6g7) [inline docs/reference/dev/auth.md]: same

File: index.d.ts [hunk 6]
```diff
@@ -65,7 +65,8 @@ interface Auth {
   signInWithRedirect: (authProvider: firebase.auth.AuthProvider) => void,
   signOut: () => void,
   unlink: (provider: string) => firebase.User,
-  updatePassword: (password: string) => void
+  updatePassword: (password: string) => void,
+  deleteProfile: () => void
```
Comment (n6g7) [inline index.d.ts]: same

File: src/auth.js [hunk 7]
```diff
@@ -124,6 +124,11 @@ function * updateProfile (profile) {
   return yield call([auth.currentUser, auth.currentUser.updateProfile], profile)
 }
 
+function * deleteProfile () {
+  const auth = this.app.auth()
+  return yield call([auth.currentUser, auth.currentUser.delete])
+}
```
Comment (n6g7) [inline src/auth.js]: same

File: README.md (hunk 4 above)
Comment (sarovin) [reply README.md]: of course

File: src/auth.test.js (hunk 1 above)
Comment (sarovin) [reply src/auth.test.js]: Oops, thanks

File: README.md [hunk 8]
```diff
@@ -40,9 +40,9 @@ You can now use `reduxSagaFirebase` methods in your sagas:
 ```js
 function* syncSaga() {
   yield fork(
-    reduxSagaFirebase.database.sync,
-    'todos',
-    { successActionCreator: syncTodos }
+  	reduxSagaFirebase.database.sync, 
+  	'todos', 
+  	{ successActionCreator: syncTodos }
```
Comment (n6g7) [inline README.md]: can you undo those changes please?

File: src/auth.js [hunk 9]
```diff
@@ -145,5 +150,6 @@ export default {
   unlink,
   updateEmail,
   updatePassword,
-  updateProfile
+  updateProfile,
+  deleteProfile
```
Comment (n6g7) [inline src/auth.js]: can you keep the alphabetical order please? :)

File: src/index.js [hunk 10]
```diff
@@ -36,7 +36,8 @@ class ReduxSagaFirebase {
       unlink: auth.unlink.bind(this),
       updateEmail: auth.updateEmail.bind(this),
       updatePassword: auth.updatePassword.bind(this),
-      updateProfile: auth.updateProfile.bind(this)
+      updateProfile: auth.updateProfile.bind(this),
+      deleteProfile: auth.deleteProfile.bind(this)
```
Comment (n6g7) [inline src/index.js]: can you keep the alphabetical order please? :)

File: src/index.test.js [hunk 11]
```diff
@@ -36,6 +36,7 @@ describe('ReduxSagaFirebase', () => {
       expect(rsf.auth.updateEmail).toBeInstanceOf(Function)
       expect(rsf.auth.updatePassword).toBeInstanceOf(Function)
       expect(rsf.auth.updateProfile).toBeInstanceOf(Function)
+      expect(rsf.auth.deleteProfile).toBeInstanceOf(Function)
```
Comment (n6g7) [inline src/index.test.js]: can you keep the alphabetical order please? :)
